packages/language-server/src/logger.ts

```diff
@@ -1,44 +1,72 @@
-import { cloneDeep } from "lodash";
+import { cloneDeep, map } from "lodash";
 import omitDeep from "omit-deep-lodash";
+import { IChildLogger, IVSCodeExtLogger } from "@vscode-logging/types";
+import { getExtensionLogger, LogLevel } from "@vscode-logging/logger";
 
-// TODO: This interface should be provided by the logging library.
-// Initially we are only providing a fairly simple API, additional variants can be added as needed...
-export interface ILogger {
-  /* eslint-disable @typescript-eslint/no-explicit-any -- meta is an object with string type key and any type value */
-  fatal: (msg: string, meta?: Record<string, any>) => void;
-  error: (msg: string, meta?: Record<string, any>) => void;
-  warn: (msg: string, meta?: Record<string, any>) => void;
-  info: (msg: string, meta?: Record<string, any>) => void;
-  debug: (msg: string, meta?: Record<string, any>) => void;
-  trace: (msg: string, meta?: Record<string, any>) => void;
-  /* eslint-enable @typescript-eslint/no-explicit-any -- meta is an object with string type key and any type value */
+// eslint-disable-next-line @typescript-eslint/no-var-requires -- Using `require` for .json file as this gets bundled with webpack correctly.
+const meta = require("../../package.json");
+
+export type ILogger = Omit<IChildLogger, "getChildLogger">;
+
+let logLevel: LogLevel = "error";
+/**
+ * We are using the VSCode Logging library right now as it is:
+ * 1. The only one of our logging libraries available on npmjs.com (currently).
+ * 2. Supports console logging which would be re-directed to the VSCode extension's output Channel
+ *    - Assuming this LSP server processes was spawned from the VSCode Extension
```

Comment:
Even if it wasn't spawned from the vscode extension, it will write the logs to the console which is the best we can do (at least until we print to a file)